client/src/main/java/org/eclipse/hono/client/Command.java

```diff
@@ -123,23 +129,25 @@ public static Command from(
         }
 
         String originalReplyToId = null;
-
+        boolean replyToLegacyEndpointUsed = false;
         if (message.getReplyTo() != null) {
             try {
                 final ResourceIdentifier replyTo = ResourceIdentifier.fromString(message.getReplyTo());
-                if (!CommandConstants.COMMAND_ENDPOINT.equals(replyTo.getEndpoint())) {
+                if (!CommandConstants.isNorthboundCommandResponseEndpoint(replyTo.getEndpoint())) {
                     // not a command message
                     validationErrorJoiner.add("reply-to not a command address: " + message.getReplyTo());
                 } else if (!tenantId.equals(replyTo.getTenantId())) {
                     // command response is targeted at wrong tenant
                     validationErrorJoiner.add("reply-to not targeted at tenant " + tenantId + ": " + message.getReplyTo());
                 } else {
+                    replyToLegacyEndpointUsed = CommandConstants.NORTHBOUND_COMMAND_LEGACY_ENDPOINT.equals(replyTo.getEndpoint());
```

Comment:
since we already have a *CommandConstants.isNorthboundCommandResponseEndpoint* method, IMHO it would be nice if we could replace this explicit comparison with an invocation of *CommandConstants.isNorthboundLegacyEndpoint*. WDYT?

Reply:
Ok, I've added and used a `isNorthboundCommandLegacyEndpoint` method.